nav2_loopback_sim/nav2_loopback_sim/loopback_simulator.py

```diff
@@ -209,6 +230,133 @@ def debug(self, msg):
         self.get_logger().debug(msg)
         return
 
+    def getMap(self):
+        request = GetMap.Request()
+        if self.map_client.wait_for_service(timeout_sec=5.0):
+            # Request to get map
+            future = self.map_client.call_async(request)
+            rclpy.spin_until_future_complete(self, future)
+            if future.result() is not None:
+                self.map = future.result().map
+                self.get_logger().info('Laser scan will be populated')
+            else:
+                self.get_logger().warn('Failed to get map')
+                self.get_logger().info('Laser scan will be populated using max range')
+        else:
+            self.get_logger().warn('Failed to get map')
+            self.get_logger().info('Laser scan will be populated using max range')
+
+    def getLaserPose(self):
+        try:
+            if self.initial_pose is None:
+                return 0.0, 0.0, 0.0
+
+            # Wait for transform and lookup
+            now = rclpy.time.Time()
+            transform = self.tf_buffer.lookup_transform(
+                'map', self.scan_frame_id, now, rclpy.duration.Duration(seconds=0.1))
+
+            # Extract pose information
+            x = transform.transform.translation.x
+            y = transform.transform.translation.y
+            rotation = transform.transform.rotation
+            theta = tf_transformations.euler_from_quaternion([
+                rotation.x,
+                rotation.y,
+                rotation.z,
+                rotation.w
+            ])[2]
+
+            return x, y, theta
+
+        except Exception as ex:
+            self.get_logger().error('Transform lookup failed: %s' % str(ex))
+            return 0.0, 0.0, 0.0
+
+    def getLaserScan(self, num_samples, x, y, theta):
+        if self.map is None or self.initial_pose is None:
+            self.scan_msg.ranges = [self.scan_msg.range_max - 0.01] * num_samples
+            return
+
+        mx, my = worldToMap(x, y, self.map)
+
+        if not mx and not my:
+            self.scan_msg.ranges = [self.scan_msg.range_max - 0.01] * num_samples
+            return
+
+        for i in range(int(num_samples)):
+            curr_angle = theta + self.scan_msg.angle_min + i * self.scan_msg.angle_increment
+            self.scan_msg.ranges[i] = self.findMapRange(mx, my, x, y, curr_angle)
```

Comment:
Should this be `ranges.append()` instead?

Reply:
both can be done actually I have initialized `self.scan_msg.ranges = [0.0] * num_samples` already and in earlier commits it was not returning anything when the scan range is less than min range do by default it was `0.0`

but later I have changed it to return `0.0`, So now `.append()` will also work

but let me know should I change it to `.append()`

Reply:
Oh ok! That works, I missed that